src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -516,7 +517,11 @@ export class KycService {
         break;
 
       case IdentShortResult.REVIEW:
-        await this.kycStepRepo.update(...kycStep.externalReview(dto));
+        if (
+          ![KycStepStatus.INTERNAL_REVIEW, KycStepStatus.MANUAL_REVIEW].includes(kycStep.status) &&
+          kycStep.getResult() !== dto
+        )
```

Comment:
`kycStep.getResult() !== dto` will not work (is comparison by reference, not by value)
See https://www.freecodecamp.org/news/javascript-comparison-operators-how-to-compare-objects-for-equality-in-js/